docs/notebooks/active_learning.pct.py

```diff
@@ -0,0 +1,177 @@
+# %% [markdown]
+# # Active Learning
+
+# %% [markdown]
+# Sometimes rather than optimizing a black-box function, we want to learn the function better by reducing our model uncertainty. The one way to do this is by doing sampling in unknown inputs regions (active learning). This notebook demonstrates to perform Bayesian active learning using `trieste`.
+
+# %%
+# %matplotlib inline
+import numpy as np
+import tensorflow as tf
+import pandas as pd
+
+np.random.seed(1793)
+tf.random.set_seed(1793)
+
+# %% [markdown]
+# ## Describe the problem
+#
+# In this example, we will consider doing active learning for the log branin function.
+
+
+# %%
+from trieste.utils.objectives import branin
+from util.plotting_plotly import plot_function_plotly
+from trieste.space import Box
+
+
+def log_branin(x):
+    return tf.math.log(branin(x))
+
+
+search_space = Box([0, 0], [1, 1])
+
+fig = plot_function_plotly(log_branin, search_space.lower, search_space.upper, grid_density=20)
+fig.update_layout(height=400, width=400)
+fig.show()
+
+# %% [markdown]
+# We begin our Bayesian active learning steps after evaluating two initial points sampled by Halton sequence.
```

Comment:
We begin our Bayesian active learning from a two-point initial design built from a Halton sequence space-filling design.